### ISSUE TITLE
detection: conversion between monai box and nndetection box

### ISSUE THREAD

**Issue by vankhoa21991** [Project-MONAI/tutorials#1225]
Hello,

In the detection example, you used the world coordinates and convert them to 'standard box' with the order of (xyzxyz). However, those xyz values are not similar to the box generated in nndetection (after preprocessed). Do you know why we have this difference? and can we have an augmentation that map the world coord to nndetection box?

Thanks

**Comment by Can-Zhao** [comment]
Thank you! nnDetection has order (xyxyzz). We feel it is unnatural.
If you would like to use nnDetection order, there is a function to convert it.

https://github.com/Project-MONAI/MONAI/blob/66422f805326a1034e040a9eacc2c8220717052e/monai/data/box_utils.py#L497
You can conver it by:
`convert_box_mode(boxes=boxes, src_mode="xyzxyz", dst_mode="xyxyzz")`

It has a Transform version:
https://github.com/Project-MONAI/MONAI/blob/66422f805326a1034e040a9eacc2c8220717052e/monai/apps/detection/transforms/dictionary.py#L98

You can do it by:
```
data = {"boxes": torch.ones(10,6)}
box_converter = ConvertBoxModed(box_keys=["boxes"], src_mode="xyzxyz", dst_mode="xyxyzz")
box_converter(data)
```

**Comment by vankhoa21991** [comment]
Thanks for your answer, and how can we map the box after post processed (world coord) to nndetection box (index coord)?

**Comment by vankhoa21991** [comment]
For example, the box in nndet for the serie 1.3.6.1.4.1.14519.5.2.1.6279.6001.333145094436144085379032922488 

is [ 58, 355,  63, 367, 181, 192]] 

for a nodule with world coord [[-37.087493896484375,
   94.93124389648438,
   -232.5,
   12.597277731716483,
   12.597277731716483,
   12.597277731716483]],

Using this world coord to convert to nndet box using these augmentations:

image_key = 'image'
box_key='box'
label_key='label'
gt_box_mode='cccwhd'
affine_lps_to_ras = True
compute_dtype = torch.float16
intensity_transform = ScaleIntensityRanged(
        keys=["image"],
        a_min=-1024,
        a_max=300.0,
        b_min=0.0,
        b_max=1.0,
        clip=True,
    )

val_transforms = Compose(
        [
            LoadImaged(keys=[image_key], meta_key_postfix="meta_dict"),
            EnsureChannelFirstd(keys=[image_key]),
            EnsureTyped(keys=[image_key, box_key], dtype=torch.float32),
            EnsureTyped(keys=[label_key], dtype=torch.long),
            Orientationd(keys=[image_key], axcodes="RAS"),
            intensity_transform,
            ConvertBoxToStandardModed(box_keys=[box_key], mode=gt_box_mode),
            AffineBoxToImageCoordinated(
                box_keys=[box_key],
                box_ref_image_keys=image_key,
                image_meta_key_postfix="meta_dict",
                affine_lps_to_ras=affine_lps_to_ras,
            ),
            EnsureTyped(keys=[image_key, box_key], dtype=compute_dtype),
            EnsureTyped(keys=label_key, dtype=torch.long),
        ]
    )

GIves an output [317.0000, 142.0000,  57.4688, 333.0000, 158.0000,  62.5312] which is different from the nndet coordinates

**Comment by vankhoa21991** [comment]
I found out the reason, because the orientation of the image was changed before making the box affine, so the box conversion is also altered.

**Comment by Can-Zhao** [comment]
Great!
